packages/app-project/src/shared/components/SubjectPicker/SubjectPicker.js

```diff
@@ -0,0 +1,157 @@
+    text-transform: uppercase;
+  }
+  th svg {
+    stroke: ${props => props.theme.global.colors['dark-1']};
+  }
+`
+
+const PAGE_SIZE = 100
+
+export default function SubjectPicker({ baseUrl, subjectSet, workflow }) {
+  const [ rows, setRows ] = useState([])
+  const [ query, setQuery ] = useState('')
+  const [ sortField, setSortField ] = useState('subject_id')
+  const [ sortOrder, setSortOrder ] = useState('asc')
+  const { indexFields } = subjectSet.metadata
+  const customHeaders = indexFields ? indexFields.split(',') : ['date', 'title', 'creators']
+
+  async function fetchSubjectData() {
+    const subjects = await fetchSubjects('15582', query, sortField, sortOrder, PAGE_SIZE)
+    const rows = await fetchRows(subjects, workflow, PAGE_SIZE)
+    setRows(rows)
+  }
+
+  useEffect(function onChange() {
+    fetchSubjectData()
+  }, [query, sortField, sortOrder])
+
+  function search(data) {
+    const query = searchParams(data)
+    setQuery(query)
+  }
+
+  function sort(data) {
+    const { property: sortField, direction: sortOrder } = data
+    if (sortField === 'status') {
+      return true;
+    }
+    setSortField(sortField)
+    setSortOrder(sortOrder)
+  }
+
+  const background = {
+    header: "accent-2",
+    body: ["white", "light-1"]
+  }
+  const pad = {
+    header: "xsmall",
+    body: "xsmall"
+  }
+
+  /*
+    Vertical spacing for the picker instructions.
+    The theme's named margins are set in multiples of 10px, so set 15px explicitly.
+  */
+  const textMargin = {
+    top: '15px',
+    bottom: 'medium'
+  }
+  return (
+    <>
+      <StyledHeading
+        level={3}
+        margin={{ top: 'xsmall', bottom: 'none' }}
+      >
+        {counterpart('SubjectPicker.heading')}
+      </StyledHeading>
+      <Paragraph
+        margin={textMargin}
+      >
+        {counterpart('SubjectPicker.byline')}
+      </Paragraph>
+      <StyledBox
+        background='brand'
+        fill
+        pad={{
+          top: '5px',
+          bottom: 'none',
+          horizontal: 'none'
+        }}
+      >
+        <Paragraph
+          color="white"
+          margin='none'
+          textAlign='center'
+        >
+          <SpacedText
+            margin='medium'
+            weight='bold'
+          >
+            {subjectSet.display_name}
+          </SpacedText>
+        </Paragraph>
+      </StyledBox>
+      <SubjectDataTable
+        background={background}
+        columns={columns(customHeaders, baseUrl)}
+        data={rows}
+        fill
+        onSearch={debounce(search, 500)}
+        onSort={sort}
+        pad={pad}
+        pin
+        replace
+        sortable
+        step={PAGE_SIZE}
+      />
+    </>
+  )
+}
+
+SubjectPicker.defaultProps = {
+  subjectSet: {
+    id: '15582',
+    display_name: 'Anti-Slavery Letters: 1800-1839',
+    metadata: {
+      indexFields: 'date,title,creators'
+    }
+  },
+  workflow: {
+    id: '5329',
+    display_name: 'Transcribe Text (Main Workflow)'
+  }
+}
```

Comment:
I'm not sure how I feel about a component having "example data" as its default values, instead of "blank data". This is my brain right now:

Bad: using "example data" as the default might lead to wonky debugging situations. e.g. subject set fails to load, and suddenly we're told it loaded "Anti-Slavery Letters: 1800-1839"

Good: "example data" provides a good template of how to use the component.

Bad: if you want to use "example data", do it in a component Story instead.

https://www.youtube.com/watch?v=XlcFTbgoeBk

Reply:
I’d forgotten about these. They were set up for a demo of the subject picker in December. 😲

Reply:
Removed now. Also, `defaultProps` is deprecated for functional components anyway.